webapp/src/client.ts

```diff
@@ -84,100 +84,133 @@ export default class CallsClient extends EventEmitter {
         }
+    }
 
-        let defaultOutputDevice: {deviceId: string; label?: string} = {
+    private getSelectedAudioDevice(deviceType: string) {
+        let selectedDevice: {deviceId: string; label?: string} = {
             deviceId: '',
         };
-        const defaultAudioOutputData = window.localStorage.getItem(STORAGE_CALLS_DEFAULT_AUDIO_OUTPUT_KEY);
-        if (defaultAudioOutputData) {
+
+        const deviceKey = deviceType === 'input' ? STORAGE_CALLS_DEFAULT_AUDIO_INPUT_KEY : STORAGE_CALLS_DEFAULT_AUDIO_OUTPUT_KEY;
+
+        const data = window.localStorage.getItem(deviceKey);
+
+        if (data) {
             try {
-                defaultOutputDevice = JSON.parse(defaultAudioOutputData);
+                selectedDevice = JSON.parse(data);
             } catch {
                 // Backwards compatibility case when we used to store the device id directly (before MM-63274).
-                defaultOutputDevice = {
-                    deviceId: defaultAudioOutputData,
+                selectedDevice = {
+                    deviceId: data,
                 };
             }
         }
 
-        if (defaultInputDevice.deviceId && !this.currentAudioInputDevice) {
-            let devices = this.audioDevices.inputs.filter((dev) => {
-                return dev.deviceId === defaultInputDevice.deviceId || dev.label === defaultInputDevice.label;
-            });
+        if (!selectedDevice.deviceId) {
+            return null;
+        }
 
-            if (devices.length > 1) {
-                // If there are multiple devices with the same label, we select the default device by ID.
-                logInfo('multiple audio input devices found with the same label, checking by id', devices);
-                devices = devices.filter((dev) => dev.deviceId === defaultInputDevice.deviceId);
-            }
+        let devices = deviceType === 'input' ? this.audioDevices.inputs : this.audioDevices.outputs;
+        devices = devices.filter((dev) => {
+            return dev.deviceId === selectedDevice.deviceId || dev.label === selectedDevice.label;
+        });
 
-            if (devices && devices.length === 1) {
-                logDebug(`found default audio input device to use: ${devices[0].label}`);
-                audioOptions.deviceId = {
-                    exact: devices[0].deviceId,
-                };
-                this.currentAudioInputDevice = devices[0];
-            } else {
-                logDebug('audio input device not found');
-                window.localStorage.removeItem(STORAGE_CALLS_DEFAULT_AUDIO_INPUT_KEY);
-            }
+        if (devices.length > 1) {
+            // If there are multiple devices with the same label, we select the selected device by ID.
+            logInfo(`getSelectedAudioDevice: multiple audio ${deviceType} devices found with the same label, checking by id`, devices);
+            devices = devices.filter((dev) => dev.deviceId === selectedDevice.deviceId);
```

Comment:
Could we use a find instead of a filter? It fits the intention better than a filter + check.